includes/functions.php

```diff
@@ -1993,3 +1993,50 @@ function recordSnmpStatistic($stat, $start_time)
     $snmp_stats["${stat}_sec"] += $runtime;
     return $runtime;
 }
+
+/**
+ * @param $device
+ * @param bool $record_perf
+ * @return array
+ */
+function device_is_up($device, $record_perf = false)
+{
+    global $config;
+
+    $address_family = snmpTransportToAddressFamily($device['transport']);
+
+    $ping_response = isPingable($device['hostname'], $address_family, $device['attribs']);
+
+    $device_perf              = $ping_response['db'];
+    $device_perf['device_id'] = $device['device_id'];
+    $device_perf['timestamp'] = array('NOW()');
+
+    if ($record_perf === true && can_ping_device($device['attribs']) === true && is_array($device_perf)) {
+        dbInsert($device_perf, 'device_perf');
+    }
+
+    $response             = array();
+    $device['pingable']   = $ping_response['result'];
```

Comment:
This will not update the global $device array(), just replace `$device['pingable']` below with `$ping_response['result']` and drop this line.